enhancements/olm/safe-cluster-upgrades.md

```diff
@@ -0,0 +1,270 @@
+
+- The CSV is missing the `operators.coreos.com/maxOpenShiftVersion` annotation
+- The value associated with the `operators.coreos.com/maxOpenShiftVersion` is not a valid Semantic Version (Valid Example: x.y.z).
+
+OLM will report the following upgradeable conditions based on the collection of CSV present on cluster:
+
+| Key                  | Meaning                                                                                            |
+| ---------------------| -------------------------------------------------------------------------------------------------- |
+| Upgradeable    CSV   | A CSV with a valid upgradeable annotation that will run on the next Minor version of OpenShift     |
+| Non-Upgradeable  CSV | A CSV with a valid upgradeable annotation that will not run on the next Minor version of OpenShift |
+| Undeterminable CSV   | A CSV that does not include a valid upgradeable annotation                                         |
+
+| Upgradeable CSV Present | Indeterminate CSV Present | Non-Upgradeable CSV Present | Status          | Message                                                                                            |
+| ----------------------- | ------------------------- | --------------------------- |---------------- | -------------------------------------------------------------------------------------------------- |
+| Yes                     | No                        | No                          | Upgradeable     | Ready for upgrade                                                                                  |
+| Don't Care              | Yes                       | No                          | Upgradeable     | The following operators may not run on the next OpenShift Version: namespace/foo, namespace/bar"   |
+| Don't Care              | Don't Care                | Yes                         | Not Upgradeable | The following operators will not run on the next OpenShift Version: namespace/foo, namespace/bar"  |
+
+#### Pros
+
+- This approach provides allows Operator Authors to specify the exact versions of OpenShift that their operator supports.
+
+#### Cons
+
+- This makes upgrade safety opt-in for each Operator, meaning that OLM cannot guarantee upgrade safety unless each installed operator specifies a maxOpenShiftVersion.
+- As new versions of OpenShift become available, Operator Authors must update their operator's maxOpenShiftVersion field.
+
+### User Stories
+
+#### Story 1
+
+As an OpenShift Cluster Admin, I want OLM to block cluster upgrades if it can determine that at least one operator will not run on the next Minor OpenShift version.
+
+### Risks and Mitigations
+
+- The most immediate risk with this solution is the fact that OLM will prevent the Cluster from upgrading if an OLM Managed Operator will not run on the next Minor OpenShift Version. This is easily circumvented if the Cluster Admin initiates the upgrade with the `force` command or by removing the offending Operator.
```

Comment:
`force` is a blunt instrument and also waives signature checks on the incoming release and other important things.  Can we replace the second sentence with something like:

> Admins will still be able to apply up security fixes with patch updates, regardless of OLM operators.  This gives them a safe window to either remove the minor-blocking operator or update it to a version that is compatible with the next minor OpenShift version.

And not mention forcing at all?  Or say something like "In extreme cases, admins can `force` to waive the `Upgradeable=False` guard (and all other guards) and update minor OpenShift versions anyway." if you feel like you really want to call out the safety-valve.

Reply:
We definitely have to mention it.  But your caveat about the additional implications of using `force` makes sense (though i'd say that's more relevant to product documentation than the EP).

Reply:
> This is easily circumvented by removing the offending Operator or by overriding the CVO's checks via means documented elsewhere.

It seems necessary to make it clear it's possible but not provide documentation on how to do it here.

Reply:
Addressed in the latest version of the enhancement.